### PR TITLE
Initial support of hardware RTC on MIMXRT10xxx board

### PR DESCRIPTION
**Pull request by morali**
Added support for RTC on NXP MIMXRT10xxx processor.

Support is using native nxp drivers. It enables rtc during initial configuration. Then you can set time using  "Rtc.SetSystemTime()" method.
## Motivation and Context

- Its necessary for providing accurate time

## How Has This Been Tested?
Tested using RTC sample project from nf repository.

## Types of changes

- [x] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

Signed-off-by: morali <jeremi.jasinski@gmail.com>

### CODE REVIEW

File: targets/FreeRTOS/NXP_MIMXRT1060_EVK/common/board/board.c [hunk 1]
```diff
@@ -211,3 +211,27 @@ void BOARD_USDHCClockConfiguration(void)
...
+    snvs_hp_rtc_datetime_t rtcDate;
+
+    SNVS_HP_RTC_GetDefaultConfig(&snvsRtcConfig);
+    SNVS_HP_RTC_Init(SNVS, &snvsRtcConfig);
+
+    /* Set a start date time and start RT */
+    rtcDate.year = 2019U;
+    rtcDate.month = 12U;
+    rtcDate.day = 25U;
+    rtcDate.hour = 10U;
```
Comment (MateuszKlatecki) [inline targets/FreeRTOS/NXP_MIMXRT1060_EVK/common/board/board.c]: Why do you set a default date and time for each RTC initialization. Time should be stored by RTC and after reboot we should be able to read it.

File: targets/FreeRTOS/NXP_MIMXRT1060_EVK/nanoCLR/target_board.h.in [hunk 2]
```diff
@@ -13,6 +13,44 @@
 
 #include <target_common.h>
 
+// enable SPIFFS
```
Comment (josesimoes) [inline targets/FreeRTOS/NXP_MIMXRT1060_EVK/nanoCLR/target_board.h.in]: All these HAL_nnnnn defines are use in ChibiOS HAL to enable a specifc MCU peripheral. They shouldn't be needed in NXP/FreeRTOS...

There could be equivalent defines used in NXP HAL, but then these would have to be ported to the appropriate ones.

File: targets/FreeRTOS/NXP_MIMXRT1060_EVK/common/board/board.c (hunk 1 above)
Comment (morali) [reply targets/FreeRTOS/NXP_MIMXRT1060_EVK/common/board/board.c]: You're right. I set it for testing and then I forgot to delete this.

File: targets/FreeRTOS/NXP_MIMXRT1060_EVK/nanoCLR/target_board.h.in (hunk 2 above)
Comment (morali) [reply targets/FreeRTOS/NXP_MIMXRT1060_EVK/nanoCLR/target_board.h.in]: Should I use "NF_FEATURE_RTC" variable, or enable RTC for NXP permanently?

Comment (josesimoes) [reply targets/FreeRTOS/NXP_MIMXRT1060_EVK/nanoCLR/target_board.h.in]: That should be an option for the target developer to decide.

nF requires a time base, so providing that there is a timer running and providing the sys tick, we are good. Doesn't matter if it's driven by an oscillator, watch crystal or whatever. 

Having said that, you should use the NF_FEATURE_RTC build option to be coherent with the rest of the targets.
Being a CMake option, you need to "expose" it to a build define.
That is what the target_board.h.in file is doing.
You can pickup the CMAke var HAL_USE_RTC_OPTION and map it to whatever is the appropriate define for NXP HAL.

Comment (morali) [reply targets/FreeRTOS/NXP_MIMXRT1060_EVK/nanoCLR/target_board.h.in]: Ok, I will try to fix this in upcoming days and then i will submit commit.

**Comment by josesimoes** [comment]
@morali this is now showing some conflicts because of the latest changes merged and ported from develop.

Please rebase your branch from upstream/develop-freertos-iMXRT10xxx and work out any outstanding issues.
If you don't mind me suggesting: don't let this drag for too long or you risk ending up with more changes and it will take you more time to clear all that. 😉

**Comment by josesimoes** [comment]
@morali you should use git command **rebase** instead of **merge**...

**Comment by morali** [comment]
@josesimoes I'm trying to do a rebase, but im having difficulties.

**Comment by josesimoes** [comment]
@morali I've made the rebase. Please pull it locally, build, test on a board and provide feedback.

**Comment by morali** [comment]
I'm sorry for the confusion... From now on I will be using console commands instead of GUI.

**Comment by josesimoes** [comment]
No problem!! We've all been there. 😄 
Git can be… confusing… at times. 😉 

As for UI tool for git, suggest that you find a decent tool. Git desktop and Git VS extension are OK for trivial operations but are tricky to use for more than that.

**Pull request merged**